src/components/Login.tsx

```diff
@@ -1,9 +1,135 @@
+  const signUpLink = isSignUp ? "Sign in now" : "Sign up now";
+  console.log("page = ", signBtn);
+  function handleLoginPage() {
+    setIsSignUp(prevVal => !prevVal);
+  }
+
+  function handleButtonClick() {
+    if (email.current && password.current) {
+      console.log("email = ", email.current.value);
+      console.log("password = ", password.current.value);
+      const message = checkValidEmail(email.current.value);
+      console.log("mesasge = ", message);
+      setIsErrorMessage(message);
+
+      if (message) return;
+
+      if (isSignUp) {
+        //Signup Logic
+        console.log("Entered If case");
+        createUserWithEmailAndPassword(
+          auth,
+          email.current.value,
+          password.current.value
+        )
+          .then((userCredential) => {
+            // Signed up
+            const user = userCredential.user;
+            console.log("user signed in = ", user);
+          })
+          .catch((error) => {
+            //Error Occurs
+            const errorCode = error.code;
+            const errorMessage = error.message;
+            setIsErrorMessage(errorCode + "-" + errorMessage);
+          });
+      } else {
+        //Sign in Logic
+        console.log("Entered else case");
+        signInWithEmailAndPassword(auth, email.current.value, password.current.value)
+          .then((userCredential) => {
+            // Signed in
+            const user = userCredential.user;
+            console.log("User LoggedIn = ", user);
+          })
+          .catch((error) => {
+            const errorCode = error.code;
+            const errorMessage = error.message;
+            setIsErrorMessage(errorCode + "-" + errorMessage);
+          });
+      }
+    }
+  }
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+  };
+
+  return (
+    <div className="absolute">
+      <HeaderComponent />
+      <form
+        onSubmit={handleSubmit}
+        className="absolute w-1/4 p-12 text-white bg-black top-0 mx-auto left-0 right-0 my-44 rounded-lg bg-opacity-85 "
+      >
+        <p className="text-3xl pb-4 pl-2 font-semibold">{signBtn}</p>
+        {isSignUp && (
+          <input
+            ref={name}
+            type="text"
+            placeholder="Enter Name"
+            className="p-3 my-2 w-full border-2 border-gray-400 bg-transparent rounded-md"
+          />
+        )}
+        <input
+          ref={email}
+          type="email"
+          placeholder="Enter Email"
+          className="p-3 my-2 w-full border-2 border-gray-400 bg-transparent rounded-md"
+        />
+        <input
+          ref={password}
+          type="password"
+          placeholder="******"
+          className="p-3 my-2 w-full bg-transparent border-2 border-gray-400 rounded-md"
+        />
+        {errorMessage && <p className="text-red-500 text-sm">{errorMessage}</p>}
+        <button
+          onClick={handleButtonClick}
```

Comment:
Can name the onClick hanlder better for understanding

Reply:
updated